### PR TITLE
Add strong salted password hashing using PBKDF2

### LINKED ISSUE #47: Perform Password Hashing on the Server
**Is your feature request related to a problem? Please describe.**
When the password hash from the `LoginCredentials` is just compared with that from the database, it offers almost no benefit over just storing passwords inside the database.

**Describe the solution you'd like**
Transmit the actual password inside `LoginCredentials` and hash it on the server. This should happen using a per-user salt.

**Describe alternatives you've considered**
Not having any security I guess ¯\_(ツ)_/¯

### PR DESCRIPTION
**Pull request by CyB3RC0nN0R**
Fixes #47

### CODE REVIEW

File: src/main/java/envoy/server/util/PasswordUtil.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * Project: <strong>envoy-server-standalone</strong><br>
+ * File: <strong>PasswordUtil.java</strong><br>
+ * Created: <strong>10.07.2020</strong><br>
+ * 
+ * @author Kai S. K. Engelbart
+ * @since Envoy Server Standalone v0.1-beta
+ */
+public class PasswordUtil {
+
+	private static final int	ITERATIONS				= 1000;
```
Comment (delvh) [inline src/main/java/envoy/server/util/PasswordUtil.java:24]: Does it make such a difference between 100 and 1000?

File: src/main/java/envoy/server/util/PasswordUtil.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	private static final int	KEY_LENGTH				= 64 * 8;
```
Comment (delvh) [inline src/main/java/envoy/server/util/PasswordUtil.java:25]: Why don't we just write 512?

File: src/main/java/envoy/server/util/PasswordUtil.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			String[]	parts		= stored.split(":");
+			int			iterations	= Integer.parseInt(parts[0]);
+			byte[]		salt		= fromHex(parts[1]);
+			byte[]		hash		= fromHex(parts[2]);
+
+			var		spec		= new PBEKeySpec(current.toCharArray(), salt, iterations, KEY_LENGTH);
+			var		skf			= SecretKeyFactory.getInstance(SECRET_KEY_ALGORITHM);
+			byte[]	testHash	= skf.generateSecret(spec).getEncoded();
+
+			int diff = hash.length ^ testHash.length;
```
Comment (delvh) [inline src/main/java/envoy/server/util/PasswordUtil.java:50]: That must've been the most complicated yet fastest way I've ever seen to subtract two numbers...
However, there might be security concerns about random positives. I hope you calculated them into your equation...

File: src/main/java/envoy/server/util/PasswordUtil.java (hunk 1 above)
Comment (CyB3RC0nN0R) [reply src/main/java/envoy/server/util/PasswordUtil.java:24]: 1000 seems to be accepted as the minimum recommended number of rounds. However, the right amount actually depends on the hardware you are running on (look [here](https://security.stackexchange.com/questions/3959/recommended-of-iterations-when-using-pkbdf2-sha256) for more information.

File: src/main/java/envoy/server/util/PasswordUtil.java (hunk 2 above)
Comment (CyB3RC0nN0R) [reply src/main/java/envoy/server/util/PasswordUtil.java:25]: This comes from the fact that the hash length is 8. It makes no difference during runtime, as constant folding takes place during compilation.

File: src/main/java/envoy/server/util/PasswordUtil.java (hunk 3 above)
Comment (CyB3RC0nN0R) [reply src/main/java/envoy/server/util/PasswordUtil.java:50]: The XOR of two unequal numbers will always be `!= 0`, so I don't see a problem here.

**Pull request merged**